### PR TITLE
update aoc layer to latest

### PR DESCRIPTION
**Pull request by wangzlei**
*Issue #, if available:*
update the aoc layer to the latest Lambda extensions API, add sub span in Lambda function
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lambda-api/src/main/java/io/awsobservability/opentelemetry/playground/lambdaapi/HelloLambdaHandler.java [hunk 1]
```diff
@@ -31,10 +35,20 @@
     public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent request, Context context) {
         var response = new APIGatewayProxyResponseEvent();
         response.setStatusCode(200);
-        response.setHeaders(request.getHeaders().entrySet()
```
Comment (wangzlei) [inline lambda-api/src/main/java/io/awsobservability/opentelemetry/playground/lambdaapi/HelloLambdaHandler.java:34]: Always get a NPE here, so comment out this part

Comment (anuraaga) [reply lambda-api/src/main/java/io/awsobservability/opentelemetry/playground/lambdaapi/HelloLambdaHandler.java:34]: Are you invoking via API gateway? I use this to debug whether headers are correctly propagated or not so want to keep it in.

File: lambda-api/build.gradle.kts [hunk 2]
```diff
@@ -48,6 +48,8 @@ dependencies {
     implementation("com.amazonaws:aws-lambda-java-core")
     implementation("com.amazonaws:aws-lambda-java-events")
     implementation("org.apache.logging.log4j:log4j-core")
+    implementation("org.apache.httpcomponents:httpclient:4.5.12")
+    implementation("commons-logging:commons-logging:1.1.1")
```
Comment (anuraaga) [inline lambda-api/build.gradle.kts:52]: We have log4j don't need to add commons-logging

**Review by anuraaga — COMMENTED** [review]
Don't see change to aoc layer in this PR, intentional?

File: lambda-api/build.gradle.kts (hunk 2 above)
Comment (wangzlei) [reply lambda-api/build.gradle.kts:52]: Got an error `java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory` if have no this dependency

Comment (anuraaga) [reply lambda-api/build.gradle.kts:52]: httpclient uses logging class without depending on it? That's pretty crazy!

Comment (wangzlei) [reply lambda-api/build.gradle.kts:52]: don't know the root cause, in my project I don't have to add commons-logging, but here will get error.

**Pull request merged**